runtime/autoload/health/provider.vim

```diff
@@ -497,6 +485,72 @@ function! s:check_for_pyenv() abort
   return [pyenv_path, pyenv_root]
 endfunction
 
+" Locate Python executable by running invocation and checking
+" sys.executable.
+function! s:locate_pythonx(invocation) abort
+  return s:normalize_path(system(a:invocation
+    \ . ' -c "from __future__ import print_function; import sys; print(sys.executable, end=\"\")"'))
```

Comment:
You can simplify this by using `sys.stdout.write()` instead of `print()`. That way you don't have to mess around with the `__future__` import and the `end=""` thing because `.write()` never adds an implicit line ending.

```suggestion
    \ . ' -c "import sys; sys.stdout.write(sys.executable)"'))
```

Reply:
Thanks, of course, you're right, that's a very good point :)

Reply:
Looks like `s:version_info` could also use this treatment?